ns for fx: make torch.ops.quantized.dropout mapping dynamic

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #76993 ns for fx: remove duplicated BNReLU mappings
* #76992 ns for fx: remove quantized ReLU6 from mapping
* #76991 ns for fx: remove GroupNorm from mapping
* #76990 ns for fx: clean up linear in relationship mapping
* **#76989 ns for fx: make torch.ops.quantized.dropout mapping dynamic**
* #76988 ns for fx: remove torch.ops.quantized.cat
* #76980 ns for fx: clean up convtranspose mappings
* #76978 ns for fx to backend_config_dict [2/x]: native lowering mappings
* #76958 ns for fx to backend_config_dict [1/x]: fused and qat modules

Summary:

Instead of hardcoding the relationship between `F.dropout` and `toq.dropout`,
read it from the mapping.

The mapping itself might need to be in the lowering file, but that's a separate
issue.

Test plan:

```
python test/test_quantization.py -k FXGraphMatcher
python test/test_quantization.py -k FXNumericSuite
```

Differential Revision: [D36215849](https://our.internmc.facebook.com/intern/diff/D36215849)

vkuzo committed:
ns for fx: make torch.ops.quantized.dropout mapping dynamic

Summary:

Instead of hardcoding the relationship between `F.dropout` and `toq.dropout`,
read it from the mapping.

The mapping itself might need to be in the lowering file, but that's a separate
issue.

Test plan:

```
python test/test_quantization.py -k FXGraphMatcher
python test/test_quantization.py -k FXNumericSuite
```

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215849).

jerryzh168 commented on torch/ao/ns/fx/mappings.py:
```diff
@@ -395,6 +390,7 @@ def get_base_name_to_sets_of_related_ops() -> Dict[str, Set[NSNodeTargetType]]:
     for source_to_target in (
         _lower_to_native_backend.QBIN_OP_MAPPING,
         _lower_to_native_backend.QBIN_RELU_OP_MAPPING,
+        quantization_mappings.DEFAULT_FLOAT_TO_QUANTIZED_OPERATOR_MAPPINGS,
```
maybe add a TODO comment to move this to _lower_to_native_backend?

vkuzo committed:
Update on "ns for fx: make torch.ops.quantized.dropout mapping dynamic"


Summary:

Instead of hardcoding the relationship between `F.dropout` and `toq.dropout`,
read it from the mapping.

The mapping itself might need to be in the lowering file, but that's a separate
issue.

Test plan:

```
python test/test_quantization.py -k FXGraphMatcher
python test/test_quantization.py -k FXNumericSuite
```

Differential Revision: [D36215849](https://our.internmc.facebook.com/intern/diff/D36215849)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215849).

vkuzo committed:
Update on "ns for fx: make torch.ops.quantized.dropout mapping dynamic"


Summary:

Instead of hardcoding the relationship between `F.dropout` and `toq.dropout`,
read it from the mapping.

The mapping itself might need to be in the lowering file, but that's a separate
issue.

Test plan:

```
python test/test_quantization.py -k FXGraphMatcher
python test/test_quantization.py -k FXNumericSuite
```

Differential Revision: [D36215849](https://our.internmc.facebook.com/intern/diff/D36215849)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215849).

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215849).

vkuzo committed:
Update on "ns for fx: make torch.ops.quantized.dropout mapping dynamic"


Summary:

Instead of hardcoding the relationship between `F.dropout` and `toq.dropout`,
read it from the mapping.

The mapping itself might need to be in the lowering file, but that's a separate
issue.

Test plan:

```
python test/test_quantization.py -k FXGraphMatcher
python test/test_quantization.py -k FXNumericSuite
```

Differential Revision: [D36215849](https://our.internmc.facebook.com/intern/diff/D36215849)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215849).

vkuzo committed:
Update on "ns for fx: make torch.ops.quantized.dropout mapping dynamic"


Summary:

Instead of hardcoding the relationship between `F.dropout` and `toq.dropout`,
read it from the mapping.

The mapping itself might need to be in the lowering file, but that's a separate
issue.

Test plan:

```
python test/test_quantization.py -k FXGraphMatcher
python test/test_quantization.py -k FXNumericSuite
```

Differential Revision: [D36215849](https://our.internmc.facebook.com/intern/diff/D36215849)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215849).

vkuzo committed:
Update on "ns for fx: make torch.ops.quantized.dropout mapping dynamic"


Summary:

Instead of hardcoding the relationship between `F.dropout` and `toq.dropout`,
read it from the mapping.

The mapping itself might need to be in the lowering file, but that's a separate
issue.

Test plan:

```
python test/test_quantization.py -k FXGraphMatcher
python test/test_quantization.py -k FXNumericSuite
```

Differential Revision: [D36215849](https://our.internmc.facebook.com/intern/diff/D36215849)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215849).

vkuzo committed:
Update on "ns for fx: make torch.ops.quantized.dropout mapping dynamic"


Summary:

Instead of hardcoding the relationship between `F.dropout` and `toq.dropout`,
read it from the mapping.

The mapping itself might need to be in the lowering file, but that's a separate
issue.

Test plan:

```
python test/test_quantization.py -k FXGraphMatcher
python test/test_quantization.py -k FXNumericSuite
```

Differential Revision: [D36215849](https://our.internmc.facebook.com/intern/diff/D36215849)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215849).
